src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -1369,3 +1401,375 @@ def repack_with_tag(tag: str,
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def __isub__(self, other: Set):
+        self.difference_update(other)
+        return self
+
+    def symmetric_difference_update(self, other: Set):
+        hb_set_symmetric_difference(self._hb_set, other._hb_set)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def __ixor__(self, other: Set):
+        self.symmetric_difference_update(other)
+        return self
+
+    def get_population(self) -> int:
+        return hb_set_get_population(self._hb_set)
+
+    def __len__(self) -> int:
+        return self.get_population()
+
+    def get_min(self) -> int:
+        return hb_set_get_min(self._hb_set)
+
+    def get_max(self) -> int:
+        return hb_set_get_max(self._hb_set)
+
+    def __iter__(self):
+        return SetIter(self)
+
+cdef class SetIter:
+    cdef Set s
+    cdef hb_set_t *_hb_set
+    cdef hb_codepoint_t _c
+
+    def __cinit__(self, s: Set):
+        self.s = s
+        self._hb_set = s._hb_set
+        self._c = s.INVALID_VALUE
+
+    def __next__(self) -> int:
+        ret = hb_set_next(self._hb_set, &self._c)
+        if not ret:
+            raise StopIteration
+        return self._c
+
+
+cdef class Map:
+    cdef hb_map_t* _hb_map
+
+    INVALID_VALUE = HB_MAP_VALUE_INVALID
+
+    def __cinit__(self, init = dict()):
+        self._hb_map = hb_map_create()
+        if not hb_map_allocation_successful(self._hb_map):
+            raise MemoryError()
+
+        for k,v in init.items():
+            self.set(k, v)
+
+    def __dealloc__(self):
+        hb_map_destroy(self._hb_map)
+
+    @staticmethod
+    cdef Map from_ptr(hb_map_t* hb_map):
+        """Create Map from taking ownership of a pointer."""
+
+        cdef Map wrapper = Map.__new__(Map)
+        wrapper._hb_map = hb_map
+        return wrapper
+
+    def copy(self) -> Map:
+        c = Map()
+        c._hb_map = hb_map_copy(self._hb_map)
+        return c
+
+    def clear(self):
+        hb_map_clear(self._hb_map)
+
+    def is_empty(self) -> bool:
+        return hb_map_is_empty(self._hb_map)
+
+    def __bool__(self) -> bool:
+        return not self.is_empty()
+
+    def get_population(self) -> int:
+        return hb_map_get_population(self._hb_map)
+
+    def __len__(self) -> int:
+        return self.get_population()
+
+    def is_equal(self, other: Map) -> bool:
+        return hb_map_is_equal(self._hb_map, other._hb_map)
+
+    def __eq__(self, other):
+        if type(other) != Map:
+            return NotImplemented
+        return self.is_equal(other)
+
+    def hash(self) -> int:
+        return hb_map_hash(self._hb_map)
+
+    def __hash__(self) -> int:
+        return self.hash()
+
+    def set(self, k: int, v: int):
+        hb_map_set(self._hb_map, k, v)
+        if not hb_map_allocation_successful(self._hb_map):
+            raise MemoryError()
+
```

Comment:
Do we want also `update()` like dict? May be also `keys()`, `values()` etc.?

Reply:
`update()` sounds good from a Python `dict` but we don't have API to update from a HB Dict. We don't have HB API for `keys()` and `values()`.